website/blog/2022-02-17-disggregated-coordinator.md

```diff
@@ -0,0 +1,152 @@
+- The rest of the query execution remains the same.
+
+
+## Memory Management
+![MemoryManagement](/img/blog/2022-02-17-disaggregated-coordinator/memory-management.png)
+
+The resource manager needs up to date information about memory and cpu utilization of the worker pool for resource group queuing. Currently, this information is periodically collected by the coordinator. In the disaggregated coordinator setup, resource managers receive query-level statistics from coordinator heartbeats, and memory pool information from worker heartbeats. This information is periodically polled by the coordinator to help make local decisions (i.e. queue/run a query, kill a query when the cluster is low in memory).
+
+## Resource Management
+![ResourceManagement](/img/blog/2022-02-17-disaggregated-coordinator/resource-management.png)
+
+The resource manager runs in multi-master mode. To support that, coordinators post query updates to all resource managers. The resource manager aggregates this information. The coordinator polls a resource manager to fetch up to date information about resource group usage in the cluster.
+
+## Resource Group Consistency Model
+
+Resource groups in a disaggregated coordinator setup are eventually consistent.  While this may lead to over-admission in certain scenarios, in practice this is mitigated by gating the resource group to only allow queries to run when certain freshness guarantees have been met (as opposed to the previous logic of checking every millisecond).  This may mean if the cluster’s resource managers are down, then queries may be queued in the coordinator’s resource groups.  This is to ensure coordinators don’t over-admit queries in the face of resource manager unavailability.
+
+The following flags can help one tune the cluster’s resource groups to the desired consistency:
+
+- **concurrency-threshold-to-enable-resource-group-refresh (default: 1.0)**
+   
+   Configure coordinator to wait for the next resource group update before allowing more queries to run on any given resource group, if running queries reached the configured limit.
+   
+   Default value is 1.0. It means once any resource group is running its max allowed queries, the coordinator has to wait for an update from the resource manager before allowing new queries to run on the given resource group. To achieve stronger consistency, reduce the percentage to lower value.
+   
+- **resource-group-runtimeinfo-refresh-interval (default: 100 ms)**
+
+  This configuration helps tune coordinator periodic polling intervals of cluster level resource group upsage from the resource manager.
+  
+## Discovery Service
+![discoveryservice](/img/blog/2022-02-17-disaggregated-coordinator/discovery-service.png)
+
+An embedded version of the discovery server runs on resource managers in a distributed mode. Discovery servers stay in sync by passing updates they receive to other discovery servers in the cluster.
+
+## Configuration
```

Comment:
This should also be in the docs. Basically I am just pushing back on putting documentation in blog posts. It leaves the docs incomplete and goes out of date over time.

Reply:
Can we only link to the docs for the configuration? This is the thing that is most likely to go out of date and be incorrect so it's better not to duplicate it.